QUARKUS-5613.md

```diff
@@ -0,0 +1,85 @@
+- Guides modified and related:
+  https://quarkus.io/guides/security-openid-connect-client-reference
+  https://quarkus.io/guides/security-oidc-code-flow-authentication
+  https://quarkus.io/guides/security-openid-connect-client-registration
+- TP/TD PRs:
+
+## Scope of the testing
+
+Our main testing scope is to ensure that the new response filter mechanism works as intended for various OIDC endpoints, ensuring that is possible intercept and handle responses
+from various OIDC endpoints.
+
+This complements the already available OidcRequestFilter functionality.
+
+Integration tests will be on the next module in our Test suite:    
+ security/keycloak-oidc-client-reactive-extended (https://github.com/quarkus-qe/quarkus-test-suite/tree/main/security/keycloak-oidc-client-reactive-extended).
+And the scope will be covered on areas related to:
+   - Security
+   - Logging /Tracing - Enabling TRACE level logs in properties will capture relevant info for our tests
+
+The test coverage will be verified for JVM, native mode, and Openshift.
+
+The testing include both OIDC Client (quarkus-oidc) and OIDC Client (quarkus-oidc-client) extensions.
+
+### What to Test - Scenarios
+
+#### Token Endpoint Response Filtering
+ - Verify that a custom OIDC response filter (TokenEndpointResponseFilter) is triggered immediately after a successful token refresh from Keycloak.
+ - Validation
+    * Ensure that filter logs messages contains "token have been refreshed".
+    * Verify 'OidcResponseContext' exposes the correct property grant_type=refresh_token
+    * Assert that the response includes "refresh_token"
+
+####  User Info Endpoint Response Filtering
+- Demonstrate that an OIDC response filter can intercept the User Info endpoint response, for instance verifying user claims.
+-  Validation
+    * Filter runs and logs claim details(sub,preferred_username)
+    * JSON body response includes the details.
+    * Logs show filter intercepting '/userinfo' response
+
+#### Client Registration Endpoint Response Filtering
+- Create a filter(OidcEndpoint.Type.CLIENT_REGISTRATION) intercepts responses to client registration calls.
+- Validation
+  * Registration data, for instance check filter logs newly registered clint_id or client_name
+  * Check if filter can detect a missing invalid property in the returned JSON
+  * Error handling, if Keycloak return a 401 the filter log propagate it correctly.
+
+All scenarios will be tested in both JVM and Native mode on baremetal and Openshift.
+
+
+## Getting familiar with the feature
+- Focus on exploratory testing of the Keycloak and OIDC testing modules
+- Check and looking thorough the guides:
+
+  https://quarkus.io/guides/security-openid-connect-client-reference
+
+  https://quarkus.io/guides/security-oidc-code-flow-authentication
+
+  https://quarkus.io/guides/security-openid-connect-client-registration
+
+
+## Community status of the extension/feature:
+All the extensions used in the module are stable and supported.
```

Comment:
Oidc client extension is experimental upstream https://code.quarkus.io/?e=oidc-client-registration&extension-search=oidc-client-regi so either change scope of testing or this section, thanks

Reply:
Yeah Oidc-client-registration extension is experimental I will update it on the TP, thanks

Reply:
Just a reminder, that `Experimental community status should be a showstopper for full or technology preview product support.` Is this extension required for filtering functionality? If yes, we should push for it to be stabilised (or drop the planned support). If no, then why do we need them?

Reply:
It is not required, it is just a convenient way to register OIDC clients dynamically. It doesn't have direct relation to this new response filter, it is just one of events and @jcarranzan is  not testing all the events anyway.